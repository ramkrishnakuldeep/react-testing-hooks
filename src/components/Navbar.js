import React from 'react';

import { Link } from 'react-router-dom';

function Navbar() {
    return (
        <nav>
            <div className="nav-wrapper">
                <Link to="/" className="brand-logo">Hooks Demo</Link>
                <ul id="nav-mobile" className="right hide-on-med-and-down">
                    <li><Link data-testid='nav-home' to="/">Home</Link></li>
                    <li><Link data-testid='nav-about' to="/about">About</Link></li>
                </ul>
            </div>
        </nav>
    )
}

export default Navbar
