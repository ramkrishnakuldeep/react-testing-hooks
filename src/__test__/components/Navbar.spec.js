import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import Navbar from '../../components/Navbar';
import {BrowserRouter} from 'react-router-dom';

afterEach(cleanup);

const withRouter = (component) => {
    return (
        <BrowserRouter> 
            {component}
        </BrowserRouter>
    )
}

test('load Navbar', async () => {
    const {asFragment} = render(withRouter(<Navbar />));
    expect(asFragment()).toMatchSnapshot();
});
