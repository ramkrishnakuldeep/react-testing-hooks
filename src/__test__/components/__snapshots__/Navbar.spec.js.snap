// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load Navbar 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="nav-wrapper"
    >
      <a
        class="brand-logo"
        href="/"
      >
        Hooks Demo
      </a>
      <ul
        class="right hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            data-testid="nav-home"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            data-testid="nav-about"
            href="/about"
          >
            About
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
